### DISCUSSION TITLE
options to visualize land claims with beacon rays instead of phantom glowstone/gold blocks

### DISCUSSION THREAD

**Discussion by taleden** [GriefPrevention/GriefPrevention#1903] [Ideas]
When inspecting/visualizing claim boundaries, rather than showing the player "fake" glowstone and gold blocks as markers, I'd love to have a config option to just show beacon rays in the corners instead. That way:

- the true blocks at the corners would remain visible instead of being masked by the phantom marker blocks
- the corners would be easier to spot even in complex builds or rough terrain, when the ground-level corner might be behind a hill or obstruction
- each separate claim could optionally use a different (random or configurable) beacon beam color, to more easily differentiate boundaries when there are lots of closely packed claims

**Comment by Jikoo** [comment]
Beacon beams are not something you can just send to a client, they require beacons. Same with beam coloring, requires stained glass.

**Reply by NumbGnat** [reply]
I can't believe there hasn't been more discussion or testing of how that might be able to be achieved.

I wonder what would happen if, instead of sending the user a glowstone, to replace the corner block, you sent the user a tinted glass block, with a beacon block (with attributes of being turned on already) right below it? 🤔

...and maybe sending them air blocks, from the tinted glass, to maxHeight, just to make sure the beacon was visible... 🤔

I wonder how much you have to override, to make a beacon beam show up... 🤔

**Reply by Jikoo** [reply]
Sending a beacon block, stained glass, and a beacon base defeats the purpose of OP's request as they specifically were requesting that fake blocks not be sent. It would require more fake world modifications than the current system.

If you're interested in writing an addon that does this, you can use the BoundaryVisualizationEvent to override GP's visualization process with a custom VisualizationProvider. You can find a sample project leveraging this capability to send packet-based highlights [here]( https://github.com/GPAddons/GPBlockHighlightBoundaries), and GP's default BoundaryVisualization used by its VisualizationProvider can be found [here](https://github.com/GriefPrevention/GriefPrevention/blob/5b7d6c484fd611f024ba068bd031883197dcea9a/src/main/java/com/griefprevention/visualization/impl/FakeBlockVisualization.java).

**Comment by NumbGnat** [comment]
> When inspecting/visualizing claim boundaries, rather than showing the player "fake" glowstone and gold blocks as markers, I'd love to have a config option to just show beacon rays in the corners instead.

I came here to suggest the same thing... 😎
